src/genie/libs/parser/iosxe/show_interface.py

```diff
@@ -3220,39 +3220,42 @@ def cli(self, output=None):
         else:
             out = output
 
-        result_dict = {}
+        parsed_dict = {}
 
         # Port      Name               Status       Vlan       Duplex  Speed Type
         # Gi1/2     TelenlqPOIU        notconnect   125          full    100 10/100/1000-TX
-        # Gi1/3     SE                 connected    132        a-full a-1000 10/100/1000-TX
-        # Gi1/7                        notconnect   99           auto   auto 10/100/1000-TX
-        # Gi1/10    To cft123     connected    trunk      a-full a-1000 10/100/1000-TX
-
-        p1 = re.compile(r'^(?P<interfaces>\w+\d+\/\d+)(?: +(?P<name>([\S ]+)))?'
-                        r' +(?P<status>\S+) +(?P<vlan>\S+) +(?P<duplex_code>[\S\-]+)'
-                        r' +(?P<port_speed>[\S\-]+) +(?P<type>\S+)$')
-
-        for line in out.splitlines():
-            line = line.strip()
-
-            m = p1.match(line)
-            if m:
-                group = m.groupdict()
-
-                intf_dict = result_dict.setdefault('interfaces', {}).\
-                                        setdefault(Common.convert_intf_name(group['interfaces']), {})
-
-                name_val = group['name'].strip()
-                if len(name_val)>0 :
-                    intf_dict['name'] = name_val
-
-                keys = ['status',
-                        'vlan', 'duplex_code', 'port_speed',
-                        'type']
+        # Gi1/1/0/1      FAST-HELLO         connected    4094       a-full a-1000 10/100/1000BaseTX
+        # Te1/1/2   VSL                connected    trunk        full  a-10G 10GBase-SR
+        # Po1       ES1SW80AUN6        connected    trunk      a-full  a-10G
 
-                for k in keys:
-                    intf_dict[k] = group[k]
-                continue
-
-        return result_dict
+        if out:
+            res = parsergen.oper_fill_tabular(device_output=out,
```

Comment:
@xiaoxinz-cisco Well, not too much. Working on the testcase I've found a problem using oper_fill_tabular.

Some fields of the table are right justified and other left justified. I.e:
```
 Port      Name               Status       Vlan       Duplex  Speed Type
 Gi1/3     SE                 connected    132        a-full a-1000 10/100/1000-TX

```
Here Duplex and Speed columns are right-justified.
So the parsing of duplex and speed column are wrong for some cases. In this  case Duplex will parse to "a-full a" (instead of "a-full" and Speed to "-1000" (instead of a-1000) :-(
Any way to deal with this using oper_fill_tabular? Or no way and I've to return to regex?

Reply:
Hi @salvapinyol ,

It is a limitation in the parsergen. One available solution is loop over each line and insert '|' between each value, then add an argument `delimiter='|'` in oper_fill_tabular.

It sounds slow, right? Thus, back to regex might be an efficient way.